test/NuGet.Core.Tests/NuGet.DependencyResolver.Core.Tests/RemoteDependencyWalkerTests.cs

```diff
@@ -1840,6 +1840,44 @@ public async Task WalkAsync_WithDowngradeInPrunedSubgraph_DoesNotReportDowngrade
             result.Downgrades.Should().BeEmpty();
         }
 
+        /// <summary>
+        /// A 1.0 -> D 1.0 (Central transitive)
+        ///       -> B 1.0 -> D 3.0 (Central transitive - should be ignored)
+        ///                -> C 1.0 -> D 2.0
+        /// </summary>
+        [Fact]
+        public async Task TransitiveDependenciesFromNonRootLibraries_AreIgnored()
+        {
+            var context = new TestRemoteWalkContext();
+            var provider = new DependencyProvider();
+            provider.Package("A", "1.0")
+                .DependsOn("B", "1.0")
+                .DependsOn("D", "1.0", LibraryDependencyTarget.Package, versionCentrallyManaged: true, libraryDependencyReferenceType: LibraryDependencyReferenceType.None);
+
+            provider.Package("B", "1.0")
+                .DependsOn("C", "1.0")
+                .DependsOn("D", "3.0", LibraryDependencyTarget.Package, versionCentrallyManaged: true, libraryDependencyReferenceType: LibraryDependencyReferenceType.None);
```

Comment:
Should this be centrally managed?
In the context of A, this isn't centrally managed, it's just a dependency of B.

Reply:
but not in the context of A. 

When A is getting restored, A doesn't care about central management configured by B, it just cares about the declared dependencies (and their versions) from B.